src/ads.ts

```diff
@@ -19,7 +19,7 @@ function insertAdPlaceholders(reactNodes: ReactNode[]): ReactNode[] {
     );
 
     const insertAd = (para: number, nodes: ReactNode[]): ReactNode[] =>
-        adIndices.includes(para) ? [...nodes, ad] : nodes;
+        adIndices.includes(para) ? [...nodes, ad(++para)] : nodes;
```

Comment:
I think `++para` is confusing here. It mutates a parameter before passing it as an argument to the `ad` function, but we do nothing with the newly modified value of `para`.

Mutating parameters is to be avoided as in most case won't do much (for small types such as `number`), but sometimes will affect the actual value (if the function receives an object for instance). It makes it hard from the caller point of view to understand what's happening.

I think simply passing `ad(para + 1)` will achieve the same effect without the confusion

Reply:
I didn't know `++foo` was a thing you could do in JavaScript! TIL...

@webb04 what's the reason for incrementing `para` here?

Reply:
At the time I wanted that ad to have a different key to the paragraphs, but I realise it would have still had a duplicated key.

I'll add something to generate IDs using `uuid` or something.

Reply:
Opinions may vary on how much of an issue this is, but using UUID would introduce side effects into all the functions that make use of keys somewhere down their call stack...